In [446]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt


In [447]:
companies=pd.read_csv('companies.txt',sep='\t',encoding='ansi')

In [448]:
companies.head()

permalink                    name  \
0                  /Organization/-Fame                   #fame   
1               /Organization/-Qounter                :Qounter   
2  /Organization/-The-One-Of-Them-Inc-  (THE) ONE of THEM,Inc.   
3                /Organization/0-6-Com                 0-6.com   
4       /Organization/004-Technologies        004 Technologies   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2                http://oneofthem.jp   
3                 http://www.0-6.com   
4  http://004gmbh.de/en/004-interact   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2                                  Apps|Games|Mobile  operating          NaN   
3                                        Curated Web  operating          CHN   
4                                           Software  operating          USA   

  state_code                 region           city  founded_at  
0         16                 Mumbai         Mumbai         NaN  
1         DE             DE - Other  Delaware City  04-09-2014  
2        NaN                    NaN            NaN         NaN  
3         22                Beijing        Beijing  01-01-2007  
4         IL  Springfield, Illinois      Champaign  01-01-2010

In [449]:
companies.shape

(66368, 10)

In [450]:
rounds2=pd.read_csv('rounds2.csv',sep=',',encoding='ANSI')

In [451]:
rounds2.head()

company_permalink  \
0                  /organization/-fame   
1               /ORGANIZATION/-QOUNTER   
2               /organization/-qounter   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                /organization/0-6-com   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015         10000000.0  
1                  A  14-10-2014                NaN  
2                NaN  01-03-2014           700000.0  
3                  B  30-01-2014          3406878.0  
4                  A  19-03-2008          2000000.0

In [452]:
rounds2.shape

(114949, 6)

## Problem 1: Data Cleaning 1

### How many unique companies are present in companies?

In [453]:
companies['permalink']=companies['permalink'].str.upper()
companies['company_name']=companies['permalink'].str.replace("/ORGANIZATION/","")

In [454]:
companies['company_name'].value_counts()

LIVELOCAL                        1
CITYBIZLIST                      1
SYMMETRIC-COMPUTING              1
SHOPCADE                         1
EPIGAMI                          1
HEARTWARE-INTERNATIONAL          1
TRIPWIRE                         1
WP-FAIL-SAFE                     1
AMPLIDATA                        1
CHAMPIONVILLAGE                  1
COMMERCE-SCIENCES                1
CLOUDBYTE                        1
YET-ANALYTICS-INC-               1
THINKR                           1
EMELODY-WORLDWIDE                1
IXICO                            1
GORSH                            1
NAPPINNAI                        1
PAPATV-CN                        1
INGENICO                         1
AAIPHARMA-SERVICES               1
SALESLOFT                        1
FOODJUNKY                        1
LOUDEYE                          1
LIDYANA                          1
OFFSET-OPTIONS                   1
MARDIL-MEDICAL                   1
ULYMPIX                          1
SEEBLINGS           

### How many unique companies are present in rounds2?

In [455]:
rounds2['company_permalink']=rounds2['company_permalink'].str.upper()

In [456]:
rounds2['company_name']=rounds2['company_permalink'].str.replace("/ORGANIZATION/","")

In [457]:
rounds2['company_name'].value_counts()

SOLARFLARE                    19
EXAGEN-DIAGNOSTICS            18
QUELLAN                       18
SEMPRIUS                      18
VOCALOCITY                    17
TIGO-ENERGY                   17
ENDOGASTRIC-SOLUTIONS         17
FRESENIUS-MEDICAL-CARE        16
POWERHOUSE-DYNAMICS           16
TYRO-PAYMENTS                 16
TERASCALA                     16
MOBIVERY                      16
PALANTIR-TECHNOLOGIES         16
APERTO-NETWORKS               15
INSTAMED                      15
ROSTIMA                       15
GROWLIFE                      15
CNS-RESPONSE                  15
NEXT-STEP-LIVING              15
DOCUSIGN                      14
AVIIR                         14
SOLARCITY                     14
BRAMASOL                      14
BTI-SYSTEMS                   14
COVENANT-SURGICAL-PARTNERS    14
COPROMOTE                     13
CARMA                         13
MOBEAM                        13
VISUALANT                     13
LITHIUMTECHNOLOGIES           13
          

### In the companies data frame, which column can be used as the unique key for each company? Write the name of the column.

#### Ans: Unique key for each company- column -> 'company_name' derived from 'permalink'. 

### Are there any companies in the rounds2 file which are not present in companies? Answer yes or no: Y/N

#### Ans: N. Both the companies & rounds2 file have same number(66368) of company names.

### Merge the two data frames so that all variables (columns) in the companies frame are added to the rounds2 data frame. Name the merged frame master_frame. How many observations are present in master_frame?

#### Ans: No. of observations=114949

In [458]:
master_frame=rounds2.merge(companies,on='company_name')

In [459]:
master_frame.head(3)

company_permalink                          funding_round_permalink  \
0     /ORGANIZATION/-FAME  /funding-round/9a01d05418af9f794eebff7ace91f638   
1  /ORGANIZATION/-QOUNTER  /funding-round/22dacff496eb7acb2b901dec1dfe5633   
2  /ORGANIZATION/-QOUNTER  /funding-round/b44fbb94153f6cdef13083530bb48030   

  funding_round_type funding_round_code   funded_at  raised_amount_usd  \
0            venture                  B  05-01-2015         10000000.0   
1            venture                  A  14-10-2014                NaN   
2               seed                NaN  01-03-2014           700000.0   

  company_name               permalink      name            homepage_url  \
0        -FAME     /ORGANIZATION/-FAME     #fame      http://livfame.com   
1     -QOUNTER  /ORGANIZATION/-QOUNTER  :Qounter  http://www.qounter.com   
2     -QOUNTER  /ORGANIZATION/-QOUNTER  :Qounter  http://www.qounter.com   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2  Application Platforms|Real Time|Social Network...  operating          USA   

  state_code      region           city  founded_at  
0         16      Mumbai         Mumbai         NaN  
1         DE  DE - Other  Delaware City  04-09-2014  
2         DE  DE - Other  Delaware City  04-09-2014

In [460]:
master_frame.shape

(114949, 17)

## Problem 2: Funding Type Analysis

### 1. Calculate the most representative value of the investment amount for each of the four funding types (venture, angel, seed, and private equity) and report the answers in the template provided on the previous page.

In [461]:
funding_type=master_frame[(master_frame['funding_round_type']=='private_equity') | 
             (master_frame['funding_round_type']=='seed')|
             (master_frame['funding_round_type']=='angel')|
            (master_frame['funding_round_type']=='venture')]

In [462]:
funding_type.head()

company_permalink  \
0                  /ORGANIZATION/-FAME   
1               /ORGANIZATION/-QOUNTER   
2               /ORGANIZATION/-QOUNTER   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                /ORGANIZATION/0-6-COM   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   

  funding_round_code   funded_at  raised_amount_usd           company_name  \
0                  B  05-01-2015         10000000.0                  -FAME   
1                  A  14-10-2014                NaN               -QOUNTER   
2                NaN  01-03-2014           700000.0               -QOUNTER   
3                  B  30-01-2014          3406878.0  -THE-ONE-OF-THEM-INC-   
4                  A  19-03-2008          2000000.0                0-6-COM   

                             permalink                    name  \
0                  /ORGANIZATION/-FAME                   #fame   
1               /ORGANIZATION/-QOUNTER                :Qounter   
2               /ORGANIZATION/-QOUNTER                :Qounter   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-  (THE) ONE of THEM,Inc.   
4                /ORGANIZATION/0-6-COM                 0-6.com   

             homepage_url                                      category_list  \
0      http://livfame.com                                              Media   
1  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
2  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
3     http://oneofthem.jp                                  Apps|Games|Mobile   
4      http://www.0-6.com                                        Curated Web   

      status country_code state_code      region           city  founded_at  
0  operating          IND         16      Mumbai         Mumbai         NaN  
1  operating          USA         DE  DE - Other  Delaware City  04-09-2014  
2  operating          USA         DE  DE - Other  Delaware City  04-09-2014  
3  operating          NaN        NaN         NaN            NaN         NaN  
4  operating          CHN         22     Beijing        Beijing  01-01-2007

In [463]:
suitable_type=funding_type.groupby(['funding_round_type'])['raised_amount_usd'].mean()
suitable_type

funding_round_type
angel             9.586945e+05
private_equity    7.330859e+07
seed              7.198180e+05
venture           1.174895e+07
Name: raised_amount_usd, dtype: float64

### 2. Based on the most representative investment amount calculated above, which investment type do you think is the most suitable for Teclov?Considering that Teclov wants to invest between 5 to 15 million USD per investment round, which investment type is the most suitable for it? Identify the investment type and, for further analysis, filter the data so it only contains the chosen investment type.

#### Ans: Since Teclov wants to invest between 5 to 15 million USD per investment round, most suitable investment type would be venture, which have most representative value of the investment amount as 11.7 million USD.

In [464]:
venture_data=master_frame[(master_frame['funding_round_type']=='venture')]

In [465]:
venture_data.head()

company_permalink  \
0                  /ORGANIZATION/-FAME   
1               /ORGANIZATION/-QOUNTER   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                /ORGANIZATION/0-6-COM   
5       /ORGANIZATION/004-TECHNOLOGIES   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
5  /funding-round/1278dd4e6a37fa4b7d7e06c21b3c1830            venture   

  funding_round_code   funded_at  raised_amount_usd           company_name  \
0                  B  05-01-2015         10000000.0                  -FAME   
1                  A  14-10-2014                NaN               -QOUNTER   
3                  B  30-01-2014          3406878.0  -THE-ONE-OF-THEM-INC-   
4                  A  19-03-2008          2000000.0                0-6-COM   
5                NaN  24-07-2014                NaN       004-TECHNOLOGIES   

                             permalink                    name  \
0                  /ORGANIZATION/-FAME                   #fame   
1               /ORGANIZATION/-QOUNTER                :Qounter   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-  (THE) ONE of THEM,Inc.   
4                /ORGANIZATION/0-6-COM                 0-6.com   
5       /ORGANIZATION/004-TECHNOLOGIES        004 Technologies   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
3                http://oneofthem.jp   
4                 http://www.0-6.com   
5  http://004gmbh.de/en/004-interact   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
3                                  Apps|Games|Mobile  operating          NaN   
4                                        Curated Web  operating          CHN   
5                                           Software  operating          USA   

  state_code                 region           city  founded_at  
0         16                 Mumbai         Mumbai         NaN  
1         DE             DE - Other  Delaware City  04-09-2014  
3        NaN                    NaN            NaN         NaN  
4         22                Beijing        Beijing  01-01-2007  
5         IL  Springfield, Illinois      Champaign  01-01-2010

## Problem 3: Country Analysis

### 1. Teclov wants to see the top nine countries which have received the highest total funding (across ALL sectors for the chosen investment type)
### 2. For the chosen investment type, make a data frame named top9 with the top nine countries (based on the total investment amount each country has received)

In [466]:
t9=master_frame.groupby('country_code')['raised_amount_usd'].sum()
t9=t9.sort_values(ascending=False)
t9=t9.reset_index()
t9=pd.DataFrame(t9)
t9
top9=t9.head(10)
top9

country_code  raised_amount_usd
0          USA       6.694821e+11
1          CHN       7.570357e+10
2          GBR       3.276705e+10
3          IND       2.768634e+10
4          CAN       1.842468e+10
5          RUS       1.127912e+10
6          DEU       1.001776e+10
7          ISR       9.713885e+09
8          FRA       9.059771e+09
9          NLD       5.966633e+09

### Identify the top three English-speaking countries in the data frame top9.

1. Top English-speaking country- USA
2. Second English-speaking country-GBR
3. Third English-speaking country-IND

## Problem 4: Sector Analysis 1

### 1. Extract the primary sector of each category list from the category_list column

In [467]:
cats=master_frame['category_list'].astype(str)

In [468]:
ls=[]
for i in cats:
    i=i.split('|')
    ls.append(i)
print(ls)

IOPub data rate exceeded.
The notebook server will temporarily stop sending output
to the client in order to avoid crashing it.
To change this limit, set the config variable
`--NotebookApp.iopub_data_rate_limit`.

Current values:
NotebookApp.iopub_data_rate_limit=1000000.0 (bytes/sec)
NotebookApp.rate_limit_window=3.0 (secs)



In [469]:
ls1=[]
for m in ls:
    ls1.append(m[0])
print(ls1)

['Media', 'Application Platforms', 'Application Platforms', 'Apps', 'Curated Web', 'Software', 'Games', 'Biotechnology', 'Biotechnology', 'Analytics', 'Analytics', 'Analytics', 'Analytics', 'Mobile', 'Mobile', 'Mobile', 'E-Commerce', 'Entertainment', 'Networking', 'Health and Wellness', 'Health and Wellness', 'Internet Marketing', 'Apps', 'Entertainment', 'E-Commerce', 'E-Commerce', 'E-Commerce', 'E-Commerce', 'Education', 'Education', 'Search', 'Art', 'Mobile', 'Beauty', 'Curated Web', 'Curated Web', 'Curated Web', 'Curated Web', 'Curated Web', 'Curated Web', 'Curated Web', 'Curated Web', 'Curated Web', 'Curated Web', 'Curated Web', 'Local Businesses', 'Local Businesses', 'Local Businesses', 'Local Businesses', 'Local Businesses', 'Cosmetics', 'Games', 'Games', 'Games', 'Hospitality', 'Hospitality', 'Education', 'Education', 'Health Care', 'Health Care', 'Analytics', 'Analytics', 'Software', 'Advertising', 'Advertising', 'Curated Web', 'Coffee', 'E-Commerce', 'E-Commerce', 'Enterprise

In [470]:
#master_frame=master_frame[(master_frame['funding_round_type']=='venture')]
#master_frame

In [471]:
#master_frame['category_list']=master_frame['category_list'].str.split('|')

In [472]:
master_frame.head()

company_permalink  \
0                  /ORGANIZATION/-FAME   
1               /ORGANIZATION/-QOUNTER   
2               /ORGANIZATION/-QOUNTER   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                /ORGANIZATION/0-6-COM   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   

  funding_round_code   funded_at  raised_amount_usd           company_name  \
0                  B  05-01-2015         10000000.0                  -FAME   
1                  A  14-10-2014                NaN               -QOUNTER   
2                NaN  01-03-2014           700000.0               -QOUNTER   
3                  B  30-01-2014          3406878.0  -THE-ONE-OF-THEM-INC-   
4                  A  19-03-2008          2000000.0                0-6-COM   

                             permalink                    name  \
0                  /ORGANIZATION/-FAME                   #fame   
1               /ORGANIZATION/-QOUNTER                :Qounter   
2               /ORGANIZATION/-QOUNTER                :Qounter   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-  (THE) ONE of THEM,Inc.   
4                /ORGANIZATION/0-6-COM                 0-6.com   

             homepage_url                                      category_list  \
0      http://livfame.com                                              Media   
1  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
2  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
3     http://oneofthem.jp                                  Apps|Games|Mobile   
4      http://www.0-6.com                                        Curated Web   

      status country_code state_code      region           city  founded_at  
0  operating          IND         16      Mumbai         Mumbai         NaN  
1  operating          USA         DE  DE - Other  Delaware City  04-09-2014  
2  operating          USA         DE  DE - Other  Delaware City  04-09-2014  
3  operating          NaN        NaN         NaN            NaN         NaN  
4  operating          CHN         22     Beijing        Beijing  01-01-2007

In [473]:
master_frame['primary_sector']=ls1

In [474]:
map_data=pd.read_csv('mapping.csv')

In [475]:
map_data.shape

(688, 10)

In [476]:
map_data.head(3)

category_list  Automotive & Sports  Blanks  Cleantech / Semiconductors  \
0           NaN                    0       1                           0   
1            3D                    0       0                           0   
2   3D Printing                    0       0                           0   

   Entertainment  Health  Manufacturing  News, Search and Messaging  Others  \
0              0       0              0                           0       0   
1              0       0              1                           0       0   
2              0       0              1                           0       0   

   Social, Finance, Analytics, Advertising  
0                                        0  
1                                        0  
2                                        0

### ### 2. Use the mapping file 'mapping.csv' to map each primary sector to one of the eight main sectors (Note that ‘Others’ is also considered one of the main sectors)

In [477]:
map_data=pd.melt(map_data,id_vars=['category_list'])
map_data=map_data[map_data['value']==1]

In [478]:
map_data=map_data.drop(['value'],axis=1)

In [479]:
map_data.columns=['primary_sector','main_sector']

In [480]:
map_data.head(17)

primary_sector          main_sector
8         Adventure Travel  Automotive & Sports
14               Aerospace  Automotive & Sports
45                    Auto  Automotive & Sports
46         Automated Kiosk  Automotive & Sports
47              Automotive  Automotive & Sports
57                Bicycles  Automotive & Sports
69        Boating Industry  Automotive & Sports
87                     CAD  Automotive & Sports
93                    Cars  Automotive & Sports
188                 Design  Automotive & Sports
189              Designers  Automotive & Sports
213                 Drones  Automotive & Sports
226      Electric Vehicles  Automotive & Sports
292                  Fuels  Automotive & Sports
331         Heavy Industry  Automotive & Sports
364  Industrial Automation  Automotive & Sports
409             Limousines  Automotive & Sports

In [481]:
master_frame=master_frame.merge(map_data,on='primary_sector')

In [482]:
master_frame.shape

(103397, 19)

## Problem 5: Sector Analysis 2

Now you have a data frame with each company’s main sector (main_sector) mappedto it. When we say sector analysis, we refer to one of the eight main sectors.

Also, you know the top three English speaking countries and the most suitablefunding type for Teclov. Let’s call the three countries 'Country 1', 'Country 2' and'Country 3' and the funding type 'FT'.Also, the range of funding preferred by Teclov is 5 to 15 million USD.

Now, the aim is to find out the most heavily invested main sectors in each of the three countries (for funding type FT and investments range of 5-15 M USD).
1. Create three separate data frames D1, D2 and D3 for each of the three
countries containing the observations of funding type FT falling within the 5-15
million USD range. The three data frames should contain:
    - All the columns of the master_frame along with the primary sector and the main sector
    - The total number (or count) of investments for each main sector in a separate column
    - The total amount invested in each main sector in a separate column. Using the three data frames, you can calculate the total number and amount ofinvestments in each main sector.

In [483]:
D1=pd.DataFrame

In [484]:
master_frame=master_frame=master_frame[(master_frame['funding_round_type']=='venture')]
master_frame.head(17)

company_permalink  \
0               /ORGANIZATION/-FAME   
1               /ORGANIZATION/90MIN   
2               /ORGANIZATION/90MIN   
3               /ORGANIZATION/90MIN   
10    /ORGANIZATION/ALL-DEF-DIGITAL   
17        /ORGANIZATION/AUDIOSOCKET   
18        /ORGANIZATION/AUDIOSOCKET   
19        /ORGANIZATION/AUDIOSOCKET   
23             /ORGANIZATION/BABBLE   
24             /ORGANIZATION/BABBLE   
25             /ORGANIZATION/BABBLE   
26  /ORGANIZATION/BASEL-SWITZERLAND   
27  /ORGANIZATION/BASEL-SWITZERLAND   
29             /ORGANIZATION/BEDLOO   
30             /ORGANIZATION/BEDLOO   
31             /ORGANIZATION/BEDLOO   
43         /ORGANIZATION/CHEFS-FEED   

                            funding_round_permalink funding_round_type  \
0   /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1   /funding-round/21a2cbf6f2fb2a1c2a61e04bf930dfe6            venture   
2   /funding-round/bd626ed022f5c66574b1afe234f3c90d            venture   
3   /funding-round/fd4b15e8c97ee2ffc0acccdbe1a98810            venture   
10  /funding-round/452a2342fe720285c3b92e9bd927d9ba            venture   
17  /funding-round/0959f59e3a89bc8db8c6ed9c646349fd            venture   
18  /funding-round/4e3a53eec6c609943df752206070b23f            venture   
19  /funding-round/bc0b412cf2fa7b739e4105ad756b47a4            venture   
23  /funding-round/43cc678b39be3e684bb88501af05ff9b            venture   
24  /funding-round/5fdb3dbeeb6eea0243702adc372130f3            venture   
25  /funding-round/9948a4b334b8988c2a5e00bc8b4be3ec            venture   
26  /funding-round/ae3c2e82851d1fd6f4ba59ee6d3f0eb2            venture   
27  /funding-round/b6403185d64ae92299ad625c9d99b05d            venture   
29  /funding-round/50155d27414ab5e0c24aa827281499f3            venture   
30  /funding-round/566f530f89904c91851c923431d83f33            venture   
31  /funding-round/e393b61f25149d58470db69a1d719999            venture   
43  /funding-round/adca195749ae9ace84684723fbe75e5b            venture   

   funding_round_code   funded_at  raised_amount_usd       company_name  \
0                   B  05-01-2015         10000000.0              -FAME   
1                 NaN  06-10-2015         15000000.0              90MIN   
2                 NaN  07-05-2013          5800000.0              90MIN   
3                 NaN  26-03-2014         18000000.0              90MIN   
10                  A  06-08-2014          5000000.0    ALL-DEF-DIGITAL   
17                NaN  02-05-2013           650000.0        AUDIOSOCKET   
18                  A  02-09-2009           750000.0        AUDIOSOCKET   
19                NaN  15-02-2012          1582300.0        AUDIOSOCKET   
23                  A  05-02-2009          2000000.0             BABBLE   
24                  A  12-05-2009          1250000.0             BABBLE   
25                  B  21-05-2010          3000000.0             BABBLE   
26                NaN  07-01-2002         13439178.0  BASEL-SWITZERLAND   
27                NaN  05-08-2005         55464142.0  BASEL-SWITZERLAND   
29                NaN  15-09-2012           657589.0             BEDLOO   
30                NaN  27-09-2013           412500.0             BEDLOO   
31                NaN  10-06-2013           848000.0             BEDLOO   
43                  A  26-02-2015          5000000.0         CHEFS-FEED   

                          permalink               name  \
0               /ORGANIZATION/-FAME              #fame   
1               /ORGANIZATION/90MIN              90min   
2               /ORGANIZATION/90MIN              90min   
3               /ORGANIZATION/90MIN              90min   
10    /ORGANIZATION/ALL-DEF-DIGITAL    All Def Digital   
17        /ORGANIZATION/AUDIOSOCKET        Audiosocket   
18        /ORGANIZATION/AUDIOSOCKET        Audiosocket   
19        /ORGANIZATION/AUDIOSOCKET        Audiosocket   
23             /ORGANIZATION/BABBLE             Babble   
24             /ORGANIZATION/BABBLE             Bab

In [485]:
D1=master_frame[(master_frame['country_code']=='USA')&(master_frame['raised_amount_usd']>=5000000)&(master_frame['raised_amount_usd']<=15000000)]

In [486]:
D1.head()

company_permalink  \
10   /ORGANIZATION/ALL-DEF-DIGITAL   
43        /ORGANIZATION/CHEFS-FEED   
98    /ORGANIZATION/HUFFINGTONPOST   
99    /ORGANIZATION/HUFFINGTONPOST   
134        /ORGANIZATION/MATCHMINE   

                             funding_round_permalink funding_round_type  \
10   /funding-round/452a2342fe720285c3b92e9bd927d9ba            venture   
43   /funding-round/adca195749ae9ace84684723fbe75e5b            venture   
98   /funding-round/7f05940c4d2dfecb8e50a0e5720e5065            venture   
99   /funding-round/9241ae16e08df17ebdc064e49e23035a            venture   
134  /funding-round/41ac526630da57ad6eb9d02431b17657            venture   

    funding_round_code   funded_at  raised_amount_usd     company_name  \
10                   A  06-08-2014          5000000.0  ALL-DEF-DIGITAL   
43                   A  26-02-2015          5000000.0       CHEFS-FEED   
98                   A  01-08-2006          5000000.0   HUFFINGTONPOST   
99                   B  01-09-2007          5000000.0   HUFFINGTONPOST   
134                  A  01-09-2007         10000000.0        MATCHMINE   

                         permalink                 name  \
10   /ORGANIZATION/ALL-DEF-DIGITAL      All Def Digital   
43        /ORGANIZATION/CHEFS-FEED            ChefsFeed   
98    /ORGANIZATION/HUFFINGTONPOST  The Huffington Post   
99    /ORGANIZATION/HUFFINGTONPOST  The Huffington Post   
134        /ORGANIZATION/MATCHMINE            MatchMine   

                      homepage_url                           category_list  \
10        http://alldefdigital.com                                   Media   
43        http://www.chefsfeed.com     Media|Mobile|Restaurants|Technology   
98   http://www.huffingtonpost.com                   Media|News|Publishing   
99   http://www.huffingtonpost.com                   Media|News|Publishing   
134           http://matchmine.com  Media|News|Reviews and Recommendations   

        status country_code state_code         region           city  \
10   operating          USA         CA    Los Angeles    Los Angeles   
43   operating          USA         CA    SF Bay Area  San Francisco   
98    acquired          USA         NY  New York City       New York   
99    acquired          USA         NY  New York City       New York   
134     closed          USA         MA         Boston        Needham   

     founded_at primary_sector    main_sector  
10          NaN          Media  Entertainment  
43   01-01-2012          Media  Entertainment  
98   09-05-2005          Media  Entertainment  
99   09-05-2005          Media  Entertainment  
134  01-01-2007          Media  Entertainment

In [487]:
D1_count=D1.groupby('main_sector')['raised_amount_usd'].count()

In [488]:
D1_count=pd.DataFrame(D1_count)

In [489]:
D1_count=D1_count.reset_index()

In [490]:
D1_count.columns=['main_sector','count_of_investments']

In [491]:
D1_count=D1_count.sort_values(by='count_of_investments',ascending=False)
D1_count

main_sector  count_of_investments
6                                   Others                  2923
1               Cleantech / Semiconductors                  2297
7  Social, Finance, Analytics, Advertising                  1912
5               News, Search and Messaging                  1563
3                                   Health                   908
4                            Manufacturing                   799
2                            Entertainment                   580
0                      Automotive & Sports                   167

In [492]:
D1_count['main_sector'][6]

'Others'

In [493]:
D1_sum=D1.groupby('main_sector')['raised_amount_usd'].sum()

In [494]:
D1_sum=pd.DataFrame(D1_sum)

In [495]:
D1_sum=D1_sum.reset_index()

In [496]:
D1_sum.columns=['main_sector','total_amount_invested']

In [497]:
D1_sum.sort_values(by=['total_amount_invested'],ascending=False)

main_sector  total_amount_invested
6                                   Others           2.608928e+10
1               Cleantech / Semiconductors           2.118100e+10
7  Social, Finance, Analytics, Advertising           1.668070e+10
5               News, Search and Messaging           1.379682e+10
3                                   Health           8.200859e+09
4                            Manufacturing           7.258553e+09
2                            Entertainment           5.000213e+09
0                      Automotive & Sports           1.454104e+09

In [498]:
D1=D1.merge(D1_count,on='main_sector')

In [499]:
D1=D1.merge(D1_sum,on='main_sector')

In [500]:
D1

company_permalink  \
0                /ORGANIZATION/ALL-DEF-DIGITAL   
1                     /ORGANIZATION/CHEFS-FEED   
2                 /ORGANIZATION/HUFFINGTONPOST   
3                 /ORGANIZATION/HUFFINGTONPOST   
4                      /ORGANIZATION/MATCHMINE   
5                      /ORGANIZATION/MEDIABONG   
6                /ORGANIZATION/NEWSCORPORATION   
7                        /ORGANIZATION/NOKEENA   
8                        /ORGANIZATION/NOKEENA   
9                         /ORGANIZATION/PLUMTV   
10                      /ORGANIZATION/SONGBIRD   
11                         /ORGANIZATION/VADIO   
12          /ORGANIZATION/7-STAR-ENTERTAINMENT   
13                       /ORGANIZATION/8DWORLD   
14                       /ORGANIZATION/8DWORLD   
15                   /ORGANIZATION/A-BIT-LUCKY   
16                        /ORGANIZATION/AKIMBO   
17                   /ORGANIZATION/ARCCOS-GOLF   
18                      /ORGANIZATION/ARKADIUM   
19        /ORGANIZATION/ARROYO-VIDEO-SOLUTIONS   
20        /ORGANIZATION/ARROYO-VIDEO-SOLUTIONS   
21                         /ORGANIZATION/ATARI   
22                /ORGANIZATION/BEE-CAVE-GAMES   
23             /ORGANIZATION/BLADE-GAMES-WORLD   
24                        /ORGANIZATION/BOONTY   
25             /ORGANIZATION/BREAKTIME-STUDIOS   
26        /ORGANIZATION/BROADWARE-TECHNOLOGIES   
27                       /ORGANIZATION/BUDDYTV   
28                     /ORGANIZATION/BUNCHBALL   
29               /ORGANIZATION/CATALYST-MOBILE   
...                                        ...   
11119           /ORGANIZATION/ZERO-MOTORCYCLES   
11120           /ORGANIZATION/ZERO-MOTORCYCLES   
11121           /ORGANIZATION/ZERO-MOTORCYCLES   
11122           /ORGANIZATION/ZERO-MOTORCYCLES   
11123              /ORGANIZATION/BANG-NETWORKS   
11124                    /ORGANIZATION/AIRWARE   
11125                    /ORGANIZATION/EXOSTAR   
11126                  /ORGANIZATION/FLY-BLADE   
11127                 /ORGANIZATION/JETSMARTER   
11128                   /ORGANIZATION/JETSUITE   
11129                     /ORGANIZATION/KYMETA   
11130                /ORGANIZATION/PLANET-LABS   
11131                    /ORGANIZATION/SENTERA   
11132                   /ORGANIZATION/SKYCATCH   
11133                /ORGANIZATION/SPACEFLIGHT   
11134           /ORGANIZATION/TACHYON-NETWORKS   
11135           /ORGANIZATION/TACHYON-NETWORKS   
11136                /ORGANIZATION/PROJECT-100   
11137                    /ORGANIZATION/COHERIX   
11138                 /ORGANIZATION/POWERDSINE   
11139    /ORGANIZATION/CLEAN-WAVE-TECHNOLOGIES   
11140       /ORGANIZATION/VENTURE-VEHICLES-INC   
11141                    /ORGANIZATION/GRABCAD   
11142               /ORGANIZATION/ONSHAPE-INC-   
11143                     /ORGANIZATION/CARWOO   
11144                  /ORGANIZATION/FALLBROOK   
11145                 /ORGANIZATION/STREETFIRE   
11146         /ORGANIZATION/VANDYNE-SUPERTURBO   
11147  /ORGANIZATION/ISOPUR-FLUID-TECHNOLOGIES   
11148             /ORGANIZATION/METAL-NETWORKS   

                               funding_round_permalink funding_round_type  \
0      /funding-round/452a2342fe720285c3b92e9bd927d9ba            venture   
1      /funding-round/adca195749ae9ace84684723fbe75e5b            venture   
2      /funding-round/7f05940c4d2dfecb8e50a0e5720e5065            venture   
3      /funding-round/9241ae16e08df17ebdc064e49e23035a            venture   
4      /funding-round/41ac526630da57ad6eb9d02431b17657            venture   
5      /funding-round/9282890ca87072025dc1807f400acee6            venture   
6      /funding-round/8f6d7c4592e43e91e8688ba342bffcb7            venture   
7      /funding-round/9225f2db6b1b74892d5de6a8744b94ea            venture   
8      /funding-round/fb7a617ed6d1a3203024f0f111417bfb            venture   
9      /funding-round/e5109c28c1b4899b068cfa552850c424            venture   
10     /funding-round/61869a104bbc2ea3669f9b32459029b7            venture   
11  

In [501]:
D2=pd.DataFrame

In [502]:
D2=master_frame[(master_frame['country_code']=='GBR')&
                
                (master_frame['raised_amount_usd']>=5000000)&(master_frame['raised_amount_usd']<=15000000)]

In [503]:
D2_count=D2.groupby('main_sector')['raised_amount_usd'].count()

In [504]:
D2_count=pd.DataFrame(D2_count)

In [505]:
D2_count=D2_count.reset_index()

In [506]:




D2_count.columns=['main_sector','count_of_investments']

In [507]:
D2_count.sort_values(by=['count_of_investments'],ascending=False)

main_sector  count_of_investments
6                                   Others                   143
1               Cleantech / Semiconductors                   127
7  Social, Finance, Analytics, Advertising                    98
5               News, Search and Messaging                    73
2                            Entertainment                    54
4                            Manufacturing                    42
3                                   Health                    24
0                      Automotive & Sports                    16

In [508]:
D2_sum=D2.groupby('main_sector')['raised_amount_usd'].sum()

In [509]:
D2_sum=pd.DataFrame(D2_sum)

In [510]:
D2_sum=D2_sum.reset_index()

In [511]:
D2_sum.columns=['main_sector','total_amount_invested']
D2_sum

main_sector  total_amount_invested
0                      Automotive & Sports           1.670516e+08
1               Cleantech / Semiconductors           1.136905e+09
2                            Entertainment           4.615111e+08
3                                   Health           2.145375e+08
4                            Manufacturing           3.619403e+08
5               News, Search and Messaging           6.157462e+08
6                                   Others           1.249124e+09
7  Social, Finance, Analytics, Advertising           8.218884e+08

In [512]:
D2_sum.sort_values(by=['total_amount_invested'],ascending=False)
D2_sum.sort_values(by='total_amount_invested',ascending=False)

main_sector  total_amount_invested
6                                   Others           1.249124e+09
1               Cleantech / Semiconductors           1.136905e+09
7  Social, Finance, Analytics, Advertising           8.218884e+08
5               News, Search and Messaging           6.157462e+08
2                            Entertainment           4.615111e+08
4                            Manufacturing           3.619403e+08
3                                   Health           2.145375e+08
0                      Automotive & Sports           1.670516e+08

In [513]:
D2=D2.merge(D2_count,on='main_sector')

In [514]:
D2=D2.merge(D2_sum,on='main_sector')

In [515]:
D2

company_permalink  \
0                       /ORGANIZATION/90MIN   
1                       /ORGANIZATION/90MIN   
2                   /ORGANIZATION/EUTECHNYX   
3                  /ORGANIZATION/MIND-CANDY   
4                  /ORGANIZATION/MIND-CANDY   
5                     /ORGANIZATION/MOBANGO   
6               /ORGANIZATION/NATURALMOTION   
7       /ORGANIZATION/OUTPLAY-ENTERTAINMENT   
8              /ORGANIZATION/PLAYCAST-MEDIA   
9                    /ORGANIZATION/PLAYER-X   
10                   /ORGANIZATION/PLAYER-X   
11                    /ORGANIZATION/PLAYJAM   
12                    /ORGANIZATION/PLUMBEE   
13                    /ORGANIZATION/SIMWORX   
14               /ORGANIZATION/SUPERAWESOME   
15                  /ORGANIZATION/TURBULENZ   
16                   /ORGANIZATION/ZATTIKKA   
17       /ORGANIZATION/AMINO-COMMUNICATIONS   
18                       /ORGANIZATION/MUBI   
19                       /ORGANIZATION/MUBI   
20                      /ORGANIZATION/YPLAN   
21               /ORGANIZATION/ASTLEYCLARKE   
22            /ORGANIZATION/MY-WARDROBE-COM   
23             /ORGANIZATION/OLIVER-SWEENEY   
24                /ORGANIZATION/STYLISTPICK   
25                /ORGANIZATION/STYLISTPICK   
26               /ORGANIZATION/STYLUS-MEDIA   
27       /ORGANIZATION/AUDIUM-SEMICONDUCTOR   
28                    /ORGANIZATION/MOVIRTU   
29           /ORGANIZATION/ATTRACTION-WORLD   
..                                      ...   
547                   /ORGANIZATION/VEEBEAM   
548        /ORGANIZATION/MAPLIN-ELECTRONICS   
549                   /ORGANIZATION/PELIKON   
550                   /ORGANIZATION/PELIKON   
551                    /ORGANIZATION/MYDECO   
552                  /ORGANIZATION/ION-CORE   
553            /ORGANIZATION/GIGGLING-SQUID   
554  /ORGANIZATION/MICROCOSM-COMMUNICATIONS   
555        /ORGANIZATION/PHOTONIC-MATERIALS   
556        /ORGANIZATION/PHOTONIC-MATERIALS   
557        /ORGANIZATION/PRAGMATIC-PRINTING   
558                   /ORGANIZATION/CANBURG   
559             /ORGANIZATION/SKILLS-MATTER   
560                 /ORGANIZATION/BLUEHEATH   
561                  /ORGANIZATION/MADE-COM   
562    /ORGANIZATION/GREENROAD-TECHNOLOGIES   
563    /ORGANIZATION/GREENROAD-TECHNOLOGIES   
564    /ORGANIZATION/GREENROAD-TECHNOLOGIES   
565    /ORGANIZATION/GREENROAD-TECHNOLOGIES   
566                 /ORGANIZATION/AUTOQUAKE   
567                 /ORGANIZATION/AUTOQUAKE   
568                 /ORGANIZATION/AUTOQUAKE   
569                 /ORGANIZATION/AUTOQUAKE   
570                    /ORGANIZATION/CARWOW   
571              /ORGANIZATION/CHARGEMASTER   
572         /ORGANIZATION/LIGHT-BLUE-OPTICS   
573         /ORGANIZATION/LIGHT-BLUE-OPTICS   
574                  /ORGANIZATION/PRODRIVE   
575                 /ORGANIZATION/STREETCAR   
576               /ORGANIZATION/YASA-MOTORS   

                             funding_round_permalink funding_round_type  \
0    /funding-round/21a2cbf6f2fb2a1c2a61e04bf930dfe6            venture   
1    /funding-round/bd626ed022f5c66574b1afe234f3c90d            venture   
2    /funding-round/d2fc787fbc5e4f468dff8b2c557993f1            venture   
3    /funding-round/47df01ed44d7b5916159051e5e32391e            venture   
4    /funding-round/c6a873b4cbdd7ea3d023a771bd3b2f99            venture   
5    /funding-round/a5c15d027c10bdd8b024823722940b9a            venture   
6    /funding-round/c228343f250a8e67272fd6bf38fd67c3            venture   
7    /funding-round/a4a704f394f2834cdfef60e2a5d4985b            venture   
8    /funding-round/3d4affd163bf57dc9a210141db8751c5            venture   
9    /funding-round/2e87ffe1b5abbdddf8c1951ffbb5b7bc            venture   
10   /funding-round/fbeb4c9e21b489079ace3abc75825936            venture   
11   /funding-round/dec82c4e2ba4601154e4867397819159            venture   
12   /funding-round/f8245833c00705e1ccb1ff9a194612cd            venture   
13   /funding-round/7cebd9d149048e4ec85ba21a41f6cd9a         

In [516]:
D3=pd.DataFrame

In [517]:
D3=master_frame[(master_frame['country_code']=='IND')&(master_frame['raised_amount_usd']>=5000000)&(master_frame['raised_amount_usd']<=15000000)]

In [518]:
D3_count=D3.groupby('main_sector')['raised_amount_usd'].count()

In [519]:
D3_count=pd.DataFrame(D3_count)

In [520]:
D3_count=D3_count.reset_index()

In [521]:
D3_count.columns=['main_sector','count_of_investments']

In [522]:
D3_count.sort_values(by=['count_of_investments'],ascending=False)

main_sector  count_of_investments
6                                   Others                   109
5               News, Search and Messaging                    52
2                            Entertainment                    33
7  Social, Finance, Analytics, Advertising                    32
4                            Manufacturing                    21
1               Cleantech / Semiconductors                    20
3                                   Health                    19
0                      Automotive & Sports                    13

In [523]:
D3_sum=D3.groupby('main_sector')['raised_amount_usd'].sum()

In [524]:
D3_sum=pd.DataFrame(D3_sum)

In [525]:
D3_sum=D3_sum.reset_index()

In [526]:
D3_sum.columns=['main_sector','total_amount_invested']

In [527]:
D3_sum.sort_values(by=['total_amount_invested'],ascending=False)

main_sector  total_amount_invested
6                                   Others           1.007410e+09
5               News, Search and Messaging           4.338345e+08
7  Social, Finance, Analytics, Advertising           2.905435e+08
2                            Entertainment           2.808300e+08
4                            Manufacturing           2.009000e+08
3                                   Health           1.677400e+08
1               Cleantech / Semiconductors           1.653800e+08
0                      Automotive & Sports           1.369000e+08

In [528]:
D3=D3.merge(D3_count,on='main_sector')

In [529]:
D3=D3.merge(D3_sum,on='main_sector')

In [530]:
D3.head()

company_permalink                          funding_round_permalink  \
0      /ORGANIZATION/-FAME  /funding-round/9a01d05418af9f794eebff7ace91f638   
1     /ORGANIZATION/DHRUVA  /funding-round/6035248811c9530b11bd442d9239a0b1   
2  /ORGANIZATION/GAMES2WIN  /funding-round/6b024f4906c288c66d1df966e6aeb256   
3  /ORGANIZATION/GAMES2WIN  /funding-round/b095563fd43d1e4fd16da3f4bcd040af   
4      /ORGANIZATION/POKKT  /funding-round/adb94c131e001a7438a4695d873d8dc1   

  funding_round_type funding_round_code   funded_at  raised_amount_usd  \
0            venture                  B  05-01-2015         10000000.0   
1            venture                NaN  27-11-2006          5000000.0   
2            venture                  A  29-03-2007          5000000.0   
3            venture                  B  30-03-2011          6000000.0   
4            venture                  B  03-11-2015          5000000.0   

  company_name                permalink       name              homepage_url  \
0        -FAME      /ORGANIZATION/-FAME      #fame        http://livfame.com   
1       DHRUVA     /ORGANIZATION/DHRUVA     Dhruva    http://www.dhruva.com/   
2    GAMES2WIN  /ORGANIZATION/GAMES2WIN  Games2Win  http://www.games2win.com   
3    GAMES2WIN  /ORGANIZATION/GAMES2WIN  Games2Win  http://www.games2win.com   
4        POKKT      /ORGANIZATION/POKKT      POKKT      http://www.pokkt.com   

   ...     status country_code state_code     region       city  founded_at  \
0  ...  operating          IND         16     Mumbai     Mumbai         NaN   
1  ...  operating          IND         19  Bangalore  Bangalore  01-01-1997   
2  ...  operating          IND         16     Mumbai     Mumbai  01-01-2005   
3  ...  operating          IND         16     Mumbai     Mumbai  01-01-2005   
4  ...  operating          IND         16     Mumbai     Mumbai  01-08-2012   

  primary_sector    main_sector count_of_investments  total_amount_invested  
0          Media  Entertainment                   33            280830000.0  
1          Games  Entertainment                   33            280830000.0  
2          Games  Entertainment                   33            280830000.0  
3          Games  Entertainment                   33            280830000.0  
4          Games  Entertainment                   33            280830000.0  

[5 rows x 21 columns]

### ### 1. Total number of investments (count)

In [531]:
print('Total number of investments for USA :',D1['count_of_investments'].sum())
print('Total number of investments for GBR :',D2['count_of_investments'].sum())
print('Total number of investments for IND :',D3['count_of_investments'].sum())

Total number of investments for USA : 21746005
Total number of investments for GBR : 57023
Total number of investments for IND : 18069


### 2. Total amount of investment (USD)

In [532]:
print('Total amount of investments for USA in USD :',D3_sum['total_amount_invested'].sum())
print('Total amount of investments for GBR in USD :',D3_sum['total_amount_invested'].sum())
print('Total amount of investments for IND in USD :',D3_sum['total_amount_invested'].sum())

Total amount of investments for USA in USD : 2683537552.0
Total amount of investments for GBR in USD : 2683537552.0
Total amount of investments for IND in USD : 2683537552.0


### 3. Top sector (based on count of investments)

In [533]:
print('Top Sector in USA :',D1_sum['main_sector'][6])
print('Top Sector in GBR :',D2_sum['main_sector'][6])
print('Top Sector in IND :',D3_sum['main_sector'][6])

Top Sector in USA : Others
Top Sector in GBR : Others
Top Sector in IND : Others


### 4. Second-best sector (based on count of investments)

In [534]:
print('Second-best sector in USA :',D1_count['main_sector'][1])
print('Second-best sector in GBR :',D1_count['main_sector'][1])
print('Second-best sector in IND :',D1_count['main_sector'][5])

Second-best sector in USA : Cleantech / Semiconductors
Second-best sector in GBR : Cleantech / Semiconductors
Second-best sector in IND : News, Search and Messaging


### 5. Third-best sector (based on count of investments) 

In [535]:
print('Second-best sector in USA :',D1_count['main_sector'][7])
print('Second-best sector in GBR :',D2_count['main_sector'][7])
print('Second-best sector in IND :',D3_count['main_sector'][2])

Second-best sector in USA : Social, Finance, Analytics, Advertising
Second-best sector in GBR : Social, Finance, Analytics, Advertising
Second-best sector in IND : Entertainment


### 6. Number of investments in the top sector 

In [536]:
print('Top Sector in USA =',D1_count['count_of_investments'][6])
print('Top Sector in GBR =',D1_count['count_of_investments'][6])
print('Top Sector in IND =',D1_count['count_of_investments'][6])

Top Sector in USA = 2923
Top Sector in GBR = 2923
Top Sector in IND = 2923


### 7. Number of investments in the second-best sector 

In [537]:
print('Second-best sector in USA =',D1_count['count_of_investments'][1])
print('Second-best sector in GBR =',D2_count['count_of_investments'][1])
print('Second-best sector in IND =',D2_count['count_of_investments'][5])

Second-best sector in USA = 2297
Second-best sector in GBR = 127
Second-best sector in IND = 73


### 8. Number of investments in the third-best sector 

In [538]:
print('Third-best sector in USA =',D1_count['count_of_investments'][6])
print('Third-best sector in GBR =',D2_count['count_of_investments'][6])
print('Third-best sector in IND =',D3_count['count_of_investments'][6])

Third-best sector in USA = 2923
Third-best sector in GBR = 143
Third-best sector in IND = 109


In [1]:
import pandas as pd
import numpy as np

In [13]:
df=pd.read_csv('HR_data.csv')
df.head(2)

Age          Workclass  fnlwgt   Education  Education_Num  \
0   39          State-gov   77516   Bachelors             13   
1   50   Self-emp-not-inc   83311   Bachelors             13   

        Martial_Status        Occupation    Relationship    Race    Sex  \
0        Never-married      Adm-clerical   Not-in-family   White   Male   
1   Married-civ-spouse   Exec-managerial         Husband   White   Male   

   Capital_Gain  Capital_Loss  Hours_per_week         Country  Target  
0          2174             0              40   United-States   <=50K  
1             0             0              13   United-States   <=50K

In [16]:
df['Education_Num'].nunique()

16

In [17]:
df['Education'].nunique()

16

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36527 entries, 0 to 36526
Data columns (total 15 columns):
Age               36527 non-null int64
Workclass         34448 non-null object
fnlwgt            36527 non-null int64
Education         36527 non-null object
Education_Num     36527 non-null int64
Martial_Status    36527 non-null object
Occupation        34440 non-null object
Relationship      36527 non-null object
Race              36527 non-null object
Sex               36527 non-null object
Capital_Gain      36527 non-null int64
Capital_Loss      36527 non-null int64
Hours_per_week    36527 non-null int64
Country           35871 non-null object
Target            36527 non-null object
dtypes: int64(6), object(9)
memory usage: 4.2+ MB


In [7]:
df['Target'].value_counts()

 <=50K     24720
 >50K       7841
 <=50K.     3001
 >50K.       965
Name: Target, dtype: int64

In [12]:
df['Target'].unique()

array([' <=50K', ' >50K', ' <=50K.', ' >50K.'], dtype=object)

In [10]:
((df.isna().sum())/len(df))*100

Age               0.000000
Workclass         5.691680
fnlwgt            0.000000
Education         0.000000
Education_Num     0.000000
Martial_Status    0.000000
Occupation        5.713582
Relationship      0.000000
Race              0.000000
Sex               0.000000
Capital_Gain      0.000000
Capital_Loss      0.000000
Hours_per_week    0.000000
Country           1.795932
Target            0.000000
dtype: float64